### PR TITLE
feat: implement the `CompositeTableRoute`

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

1. Implement the `CompositeTableRoute`

The `CompositeTableRoute` stores all level routes of a table:
- Stores `PhysicalTableRouteValue` for logical table.
- Stores `LogicalTableRouteValue`, `PhysicalTableRouteValue` for the logical table.

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

**Review by fengjiachun — APPROVED** [review]
LGTM

File: src/common/meta/src/cache/table/composite_table_route.rs [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+use crate::key::table_route::{LogicalTableRouteValue, PhysicalTableRouteValue};
+
+/// [CompositeTableRoute] stores all level routes of a table.
+/// - Stores [PhysicalTableRouteValue] for logical table.
+/// - Stores [LogicalTableRouteValue], [PhysicalTableRouteValue] for the logical table.
+#[derive(Clone)]
+pub enum CompositeTableRoute {
+    Physical(Arc<PhysicalTableRouteValue>),
+    Logical(Arc<LogicalTableRouteValue>, Arc<PhysicalTableRouteValue>),
+}
```
Comment (waynexia) [inline src/common/meta/src/cache/table/composite_table_route.rs:36]: What's the difference between this struct and `TableRoute`, and when to use which?

File: src/common/meta/src/cache/table/composite_table_route.rs [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+                }
+            }
+        })
+    })
+}
+
+fn invalidator<'a>(
+    cache: &'a Cache<TableId, Arc<CompositeTableRoute>>,
+    ident: &'a CacheIdent,
+) -> BoxFuture<'a, Result<()>> {
```
Comment (waynexia) [inline src/common/meta/src/cache/table/composite_table_route.rs:119]: why not write an `async fn`

Comment (WenyXu) [reply src/common/meta/src/cache/table/composite_table_route.rs:119]: The `CacheContainer` requires the signature `Box<dyn for<'a> Fn(&'a Cache<K, V>, &'a CacheToken) -> BoxFuture<'a, Result<()>> + Send + Sync>;`

File: src/common/meta/src/cache/table/composite_table_route.rs (hunk 1 above)
Comment (WenyXu) [reply src/common/meta/src/cache/table/composite_table_route.rs:36]: `TableRoute` only contains the first level table route, for logical table, it only contains `LogicalTableRouteValue`. 
We should use the `CompositeTableRoute ` when the system needs to find the `PhysicalTableRouteValue` for a logical table.

**Review by waynexia — APPROVED** [review]

**Pull request merged**